### PR TITLE
feat: JSON Files Merging Script

### PR DESCRIPTION
**Pull request by OmaymaMahjoub**
## What?
Add a function that helps in merging multiple json files that contains different algos or envs or tasks or seeds related to the same experiment setup in order to have a centralised json file that will be used in the plotting tools.

## Why? 
We often run experiments on different computes causing us to have the same benchmark experiments in different json files.

### CODE REVIEW

File: marl_eval/utils/merge_json_files.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import os
+from collections import defaultdict
+from typing import Dict, Tuple
+
+
+def _read_json_files(directory: str) -> list:
+    """Reads all JSON files in a directory and returns a list of JSON objects."""
+    json_data = []
+
+    for filename in os.listdir(directory):
```
Comment (RuanJohn) [inline marl_eval/utils/merge_json_files.py:26]: A minor detail here is that we are assuming that all the json files will be in a single directory at the same level. Would it be possible to traverse the directory and get all the files that might also be in other folders in the directory?

File: README.md [hunk 2]
```diff
@@ -142,6 +142,11 @@ In order to use the tools we suggest effectively, raw data json files are requir
...
 
+#### Data Tooling
+[**JSON Files Merging Script**](marl_eval/utils/merge_json_files.py): We offer a function called `concatenate_files` that reads json files from a specified directory, concatenates their contents into a single structured dictionary,
```
Comment (RuanJohn) [inline README.md:146]:
```suggestion
[**JSON Files Merging Script**](marl_eval/utils/merge_json_files.py): We offer a function called `concatenate_files` that can be found in `marl_eval/utils` that reads json files from a specified directory and concatenates their contents into a single structured dictionary,
```

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Thank you for this @OmaymaMahjoub ! Just some minor comments :)

File: marl_eval/utils/merge_json_files.py (hunk 1 above)
Comment (OmaymaMahjoub) [reply marl_eval/utils/merge_json_files.py:26]: yeah we can do recursive way I will update it!

File: marl_eval/utils/merge_json_files.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        return _check_seed(concatenated_data, algo_data, seed_number)
+    else:
+        return seed_number
+
+
+def concatenate_files(directory: str, json_path: str = "./concatenation") -> Dict:
+    """Concatenate all json files in a directory and save the result in a json file."""
+    # Read all json files in a directory
+    json_data = _read_json_files(directory)
+
```
Comment (RuanJohn) [inline marl_eval/utils/merge_json_files.py:67]:
```suggestion
# Create target folder
    if not os.path.exists(json_path):
        os.makedirs(json_path)
```

File: marl_eval/utils/merge_json_files.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        seed_string, seed_n = _get_seed_number(seed_number)
+        seed_number = (
+            f"{seed_string}_{seed_n+1}" if seed_string != "" else str(seed_n + 1)
+        )
+        return _check_seed(concatenated_data, algo_data, seed_number)
+    else:
+        return seed_number
+
+
+def concatenate_files(directory: str, json_path: str = "./concatenation") -> Dict:
```
Comment (RuanJohn) [inline marl_eval/utils/merge_json_files.py:63]: Ideally we would want json_path to be a directory. We could have the string end in `/`?

File: marl_eval/utils/merge_json_files.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                            concatenated_data[env_name][scenario_name][algo_name],
+                            algo_data,
+                            seed_number,
+                        )
+                        concatenated_data[env_name][scenario_name][algo_name][
+                            seed_n
+                        ] = algo_data
+
+    # Save concatenated data in a json file
+    with open(f"{json_path}.json", "w") as f:
```
Comment (RuanJohn) [inline marl_eval/utils/merge_json_files.py:89]: Then for storing, make sure `json_path` ends in `/` and give a name to the merged json file. 
Something like: 
```
if json_path[-1] != "/": 
  json_path += "/"
  
with open(f"{json_path}/metrics.json", "w") as f:
```

**Comment by RuanJohn** [comment]
Closing because the same code is in #45 and it is easier to review only one PR.

**Comment by OmaymaMahjoub** [comment]
This feature is handled in #45 therefore no need to merge this PR

**Pull request closed without merging**